guides/essential/01_setup.md

```diff
@@ -0,0 +1,243 @@
+# Essential Guide 1: Setup
+
+Welcome to the first part of the 10-step-guide of setting up a working Rails CRUD app with `matestack-ui-core`!
+
+## Introduction
+In this guide, we will
+- create a new Rails application
+- change it to use `postgresql` and `matestack-ui-core`
+- add a simple matestack app and two pages
+- deploy the application to Heroku
+
+## Prerequisites
+To follow along, make sure you have successfully installed
+- Ruby (Version > 2.6, [view installation details](https://www.ruby-lang.org))
+- RubyOnRails (Version >6.0, [view installation details](https://rubyonrails.org/))
+- Heroku CLI ([view installation details](https://devcenter.heroku.com/articles/getting-started-with-ruby#set-up))
+- Postgresql ([view installation details](https://devcenter.heroku.com/articles/heroku-postgresql#local-setup))
+
+<details>
+<summary>Note for Linux/Ubuntu users</summary>
+You may need to install additional libraries by running <br/>
+<code>sudo apt-get -y install postgresql postgresql-contrib libpq-dev</code>
+instead of only running <br/>
+<code>sudo apt-get install postgresql</code>.
+</details>
+<br/>
+
+The contents of this article are heavily inspired by [Getting Started on Heroku with Rails 6.x](https://devcenter.heroku.com/articles/getting-started-with-rails6), but goes beyond it by introducing the `matestack-ui-core` gem and setting it up with some example content. Both beginners and experienced Ruby/Rails developers should be able to follow along.
+
+## Getting started
+
+In the terminal, create a new app by running
+
+```sh
+rails new matestack-demo-application
+```
+
+and switch into the newly created project via
+
+```sh
+cd matestack-demo-application
+```
+
+To make sure things work as expected, you can run
+
+```sh
+rails s
+```
+
+to start the application - visiting [localhost:3000](http://localhost:3000/) now should reveal the canonical **Yay! You're on Rails!** screen!
+
+## Adding Postgres and MatestackUiCore
+
+In the Gemfile, replace the line starting with `gem 'sqlite3'` with `gem 'pg'`.
+Make sure to run `bundle install` afterwards and replace the contents of `config/database.yml` with
+
+```yaml
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  # For details on connection pooling, see Rails configuration guide
+  # https://guides.rubyonrails.org/configuring.html#database-pooling
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
+
+development:
+  <<: *default
+  database: myapp_development
+
+test:
+  <<: *default
+  database: myapp_test
+
+production:
+  <<: *default
+  database: myapp_production
+  username: myapp
+  password: <%= ENV['MYAPP_DATABASE_PASSWORD'] %>
+```
+
+To install `matestack-ui-core`, add `gem 'matestack-ui-core'` to your Gemfile and run `bundle install`.
```

Comment:
There's too much described inline within the next ~10 line. I would recommend to use lists or proceed step by step :)